lib/assets/javascripts/turbograft/turbohead.coffee

```diff
@@ -4,211 +4,90 @@ class window.TurboHead
-    {activeScripts, newScripts} = processScripts(activeAssets, upstreamAssets)
 
-    if hasScriptConflict(activeScripts, newScripts)
-      return failureCallback()
+    newScripts = upstreamAssets
+      .filter(filterForNodeType('SCRIPT'))
+      .filter(noMatchFor({attribute: 'src', inCollection: activeAssets}))
 
-    updateLinkTags(activeAssets, upstreamAssets)
-    updateScriptTags(@activeDocument, newScripts, successCallback)
+    newLinks = upstreamAssets
+      .filter(filterForNodeType('LINK'))
+      .filter(noMatchFor({attribute: 'href', inCollection: activeAssets}))
 
-updateLinkTags = (activeAssets, upstreamAssets) ->
-  activeLinks = activeAssets.filter(filterForNodeType('LINK'))
-  upstreamLinks = upstreamAssets.filter(filterForNodeType('LINK'))
-  remainingActiveLinks = removeStaleLinks(activeLinks, upstreamLinks)
-  reorderedActiveLinks = reorderActiveLinks(remainingActiveLinks, upstreamLinks)
-  insertNewLinks(reorderedActiveLinks, upstreamLinks)
+    if newScripts.concat(newLinks).some(hasAssetConflicts(activeAssets))
+      return failureCallback()
 
-updateScriptTags = (activeDocument, newScripts, callback) ->
-  asyncSeries(
-    newScripts.map((scriptNode) -> insertScriptTask(activeDocument, scriptNode)),
-    callback
-  )
+    updateTasks = [
+      (done) => updateLinkTags(@activeDocument, newLinks, done),
+      (done) => updateScriptTags(@activeDocument, newScripts, done)
+    ]
+
+    asyncSeries(updateTasks, successCallback)
 
 extractTrackedAssets = (doc) ->
   [].slice.call(doc.querySelectorAll('[data-turbolinks-track]'))
 
 filterForNodeType = (nodeType) ->
   (node) -> node.nodeName == nodeType
 
-hasScriptConflict = (activeScripts, newScripts) ->
-  hasExistingScriptAssetName = (upstreamNode) ->
-    activeScripts.some (activeNode) ->
-      upstreamNode.dataset.turbolinksTrackScriptAs == activeNode.dataset.turbolinksTrackScriptAs
+noMatchFor = ({attribute, inCollection}) ->
+  (node) ->
+    !inCollection.some((nodeFromCollection) -> node[attribute] == nodeFromCollection[attribute])
+
+hasAssetConflicts = (activeAssets) ->
+  (newNode) ->
+    activeAssets.some((activeNode) ->
+      trackName = newNode.dataset.turbolinksTrack
+      trackName == activeNode.dataset.turbolinksTrack &&
+      trackName != 'true'
+    )
 
-  newScripts.some(hasExistingScriptAssetName)
+updateLinkTags = (activeDocument, newLinks, callback) ->
+  # style tag load events don't work in all browsers
+  # as such we just hope they load ¯\_(ツ)_/¯
+  newLinks.forEach((linkNode) -> insertLinkTask(activeDocument, linkNode)())
+  callback()
+
+updateScriptTags = (activeDocument, newScripts, callback) ->
+  asyncSeries(
+    newScripts.map((scriptNode) -> insertScriptTask(activeDocument, scriptNode)),
+    callback
+  )
 
 asyncSeries = (tasks, callback) ->
   return callback() if tasks.length == 0
   task = tasks.shift()
   task(-> asyncSeries(tasks, callback))
 
+
+asyncParallel = (tasks, callback) ->
```

Comment:
Oops.

This is a leftover from when I was still trying to find a way to load/unload the link tags while waiting for all of their events asynchronously. It can be removed for the time being.